x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/index/RestrictedIndicesNames.java

```diff
@@ -21,10 +23,21 @@
     public static final String INTERNAL_SECURITY_TOKENS_INDEX_7 = ".security-tokens-7";
     public static final String SECURITY_TOKENS_ALIAS = ".security-tokens";
 
+    // public for tests
+    public static final String ASYNC_SEARCH_PREFIX = ".async-search-";
+    private static final CharacterRunAutomaton ASYNC_SEARCH_AUTOMATON =
+            new CharacterRunAutomaton(Automatons.patterns(ASYNC_SEARCH_PREFIX + "*"));
```

Comment:
An automaton seems like overkill for a prefix, could we just use `startsWith` in `isRestricted` ?

Reply:
Sounds fair, I made the change as suggested.